staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -0,0 +1,186 @@
+}
+
+// ToOptions converts from CLI inputs to runtime inputs.
+func (flags *WhoAmIFlags) ToOptions(ctx context.Context, args []string) (*WhoAmIOptions, error) {
+	w := &WhoAmIOptions{
+		ctx:       ctx,
+		IOStreams: flags.IOStreams,
+	}
+
+	clientConfig, err := flags.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientSet, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	w.authClient = clientSet.AuthenticationV1alpha1()
+
+	if !flags.PrintFlags.OutputFlagSpecified() {
+		w.printTable = true
+		w.printer = printers.NewTablePrinter(printers.PrintOptions{})
+	} else {
+		w.printer, err = flags.PrintFlags.ToPrinter()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return w, nil
+}
+
+// WhoAmIOptions is the start of the data required to perform the operation. As new fields are added,
+// add them here instead of referencing the cmd.Flags()
+type WhoAmIOptions struct {
+	authClient authenticationv1alpha1client.AuthenticationV1alpha1Interface
+	ctx        context.Context
+
+	printer    printers.ResourcePrinter
+	printTable bool
+
+	genericclioptions.IOStreams
+}
+
+var (
+	whoAmILong = templates.LongDesc(`
+		Experimental: Check who you are and your attributes (groups, extra).
+
+        This command is helpful to get yourself aware of the current user attributes, 
+        especially when dynamic authentication, e.g., token webhook, auth proxy, or OIDC provider, 
+        is enabled in the Kubernetes cluster.
+	`)
+
+	whoAmIExample = templates.Examples(`
+		# Get your subject attributes.
+		kubectl auth who-am-i
+		
+		# Get your subject attributes in JSON format.
+		kubectl auth who-am-i -o json
+	`)
+)
+
+// NewCmdWhoAmI returns an initialized Command for 'auth who-am-i' sub command. Experimental.
+func NewCmdWhoAmI(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	flags := NewWhoAmIFlags(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   "who-am-i",
+		DisableFlagsInUseLine: true,
+		Short:                 "Experimental: Check self subject attributes",
+		Long:                  whoAmILong,
+		Example:               whoAmIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			o, err := flags.ToOptions(cmd.Context(), args)
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	flags.AddFlags(cmd)
+	return cmd
+}
+
+// Run prints all user attributes.
+func (o WhoAmIOptions) Run() error {
+	sar := &authenticationv1alpha1.SelfSubjectAttributesReview{}
+	response, err := o.authClient.SelfSubjectAttributesReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	if !o.printTable {
+		// Explicitly set type to avoid resource printer errors
+		response.TypeMeta = metav1.TypeMeta{
+			APIVersion: "authentication.k8s.io/v1alpha1",
+			Kind:       "SelfSubjectAttributesReview",
+		}
+		return o.printer.PrintObj(response, o.Out)
```

Comment:
I thought we were able to pass typed objects to standard printers from other commands and it handled them appropriately... I'm not sure why this is needing to override typemeta. @soltysh, do you see why this command is having to do something different?

Reply:
No type setting should be necessary if you're printing object retrieved from apiserver, it's only required when that is not set, which happens only when creating resources so that we can print them with dry-run. ie. `kubectl create` commands.